deploy-service/common/src/main/java/com/pinterest/deployservice/handler/DeployHandler.java

```diff
@@ -407,7 +408,11 @@ DeployBean getLastSucceededDeploy(EnvironBean envBean) throws Exception {
                 }
             }
             index += 1;
+            filterBean.setPageIndex(index);
         }
+        LOG.warn("Latest 100000 deploys are all failed for {}. Give up", envBean.getEnv_id());
```

Comment:
Maybe change to 

LOG.warn("Latest {} deploys are all failed for {}. Give up", maxPages * size, envBean.getEnv_id());